tests/sockets_cores_threads_test.go

```diff
@@ -0,0 +1,484 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	"encoding/xml"
+	"flag"
+	"github.com/google/goexpect"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"fmt"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+
+	yaml "gopkg.in/yaml.v2"
+	tframework "kubevirt.io/kubevirt-ansible/tests/framework"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	ktests "kubevirt.io/kubevirt/tests"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type Config struct {
```

Comment:
Why not reuse the kubevirt/kubernetes API objects?

Reply:
Could you please show an example?

Reply:
^^ @MarSik ^^

Reply:
https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3144 this structure implements the same I think

Reply:
I've just did it